torch/csrc/autograd/profiler.h

```diff
@@ -131,15 +131,50 @@ struct TORCH_API Event final {
       uint16_t thread_id,
       bool record_cuda,
       at::RecordFunctionHandle handle = 0,
-      std::vector<std::vector<int64_t>>&& shapes = {})
+      std::vector<std::vector<int64_t>>&& shapes = {}, int node_id = -1)
       : name_(std::move(name)),
         kind_(kind),
         thread_id_(thread_id),
         handle_(handle),
-        shapes_(shapes) {
+        shapes_(shapes), node_id_(node_id) {
     record(record_cuda);
   }
 
+  void setNodeId(int node_id) {
+    node_id_ = node_id;
+  }
+
+  // Returns IValues corresponding to event structure, to be used for
+  // serialization.
+  at::IValue getEventIValues() const {
+    c10::impl::GenericList eventIValueList(at::AnyType::get());
+    eventIValueList.push_back(at::IValue(static_cast<int64_t>(kind_)));
+    eventIValueList.push_back(at::IValue(std::string(name_.str())));
+    eventIValueList.push_back(at::IValue(thread_id_));
+    eventIValueList.push_back(at::IValue(static_cast<double>(handle_)));
```

Comment:
why is it double?, same below in fromIValue

Reply:
It looks like the ivalue int constructors are int64_t and int32, while handle_ is typedeffed to uint64_t. So we'd have to static cast to int64_t which could cause overflow issues, which is why i used the double constructor